website/server/models/task.js

```diff
@@ -25,6 +25,21 @@ export let TaskSchema = new Schema({
   type: {type: String, enum: tasksTypes, required: true, default: tasksTypes[0]},
   text: {type: String, required: true},
   notes: {type: String, default: ''},
+  shortName: {
+    type: String,
+    match: [/^[a-zA-Z0-9-_]+$/, 'Task short names can only contain alphanumeric characters, underscores and dashes.'],
+    validate: [{
+      validator: function validateShortNameOwnedByUser () {
+        return Boolean(this.userId);
+      },
+      msg: 'Task short names can only be applied to tasks in a user\'s own task list.',
+    }, {
+      validator: function validateShortNameIsNotUuid (val) {
+        return !validator.isUUID(val);
+      },
+      msg: 'Task short names cannot be uuids.',
+    }],
+  },
```

Comment:
@Alys Was able to split these out into multiple validations, which makes a lot more sense to me.